backend/app/main.py

```diff
@@ -277,20 +277,52 @@ async def upload_assignments(
     csv_rows = d["assignments"]
     document_id = d["document_id"]
 
+    # process CSV via temp tables
+    session.execute(text("DROP TABLE IF EXISTS temploader"))
     session.execute(text("""CREATE TEMP TABLE temploader (
         geo_id TEXT,
         zone INT
     )"""))
     cursor = session.connection().connection.cursor()
     with cursor.copy("COPY temploader (geo_id, zone) FROM STDIN") as copy:
         for record in csv_rows:
-            copy.write_row([record[0], int(record[1])])
+            if record[1] == "":
+                copy.write_row([record[0], None])
+            else:
+                copy.write_row([record[0], int(record[1])])
 
+    # insert into actual assignments table
     session.execute(text("""
         INSERT INTO document.assignments (geo_id, zone, document_id)
         SELECT geo_id, zone, :document_id
         FROM temploader
     """), {"document_id":  document_id})
+
+    # find items to unshatter
+    results = session.execute(text("""
+        SELECT DISTINCT(SUBSTR(geo_id, 1, 11)) AS vtd, zone
```

Comment:
H: Unfortunately, VTD GEOIDs are not perfectly hierarchical with blocks. They diverge at the county level, so we can't rely on slicing the IDs to identify VTDs to heal/unshatter.

![image](https://github.com/user-attachments/assets/e68d14e0-bcc2-499f-b375-580dc042569e)

Reply:
:+1: that makes sense. I also realized that I was doing this in reverse (un-shattering blocks, when on a fresh map I need to be shattering the VTDs which need block-level resolution)